### PR TITLE
chore: Add missing form fields to RNA example app

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

Add missing fields from signup with attributes RNA example app.

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]

File: examples/react-native/src/ui/components/authenticator/sign-up-with-attributes/Example.tsx [hunk 1]
```diff
@@ -14,6 +14,41 @@ Amplify.configure({
...
+    gender: {
+      label: 'Gender',
+      isRequired: false,
+      type: 'text',
+    },
+    locale: {
+      label: 'Locale',
+      isRequired: false,
+      type: 'text',
+    },
```
Comment (calebpollman) [inline examples/react-native/src/ui/components/authenticator/sign-up-with-attributes/Example.tsx:48]: Not sure how much value including these in the example code provides, imo kind of bloats the example code. Are the keys not documented?

Comment (calebpollman) [reply examples/react-native/src/ui/components/authenticator/sign-up-with-attributes/Example.tsx:48]: Is this example used in the docs?

Comment (ioanabrooks) [reply examples/react-native/src/ui/components/authenticator/sign-up-with-attributes/Example.tsx:48]: No, docs use this [example](https://github.com/aws-amplify/amplify-ui/blob/main/docs/src/pages/%5Bplatform%5D/connected-components/authenticator/configuration/signUpAttributes.all.react-native.mdx)

Comment (calebpollman) [reply examples/react-native/src/ui/components/authenticator/sign-up-with-attributes/Example.tsx:48]: Got it, thanks!

**Review by calebpollman — APPROVED** [review]

**Pull request merged**